app/models/user.rb

```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```

Comment:
@KDlamini  Thank you very much for your kind review and feedback.
- After I added `:confirmable` to the devise, I faced this error while I tried to sign up. Can you help me figure it out?
![error while sign up](https://user-images.githubusercontent.com/96838030/188901643-e098aeb9-9428-4518-a94d-51a636d7eb75.png)

Reply:
and after I added the confirmable to devise migrate I get this error:

![second error](https://user-images.githubusercontent.com/96838030/188912782-baa7c2ad-439b-4c69-be2a-1d54872a2918.png)

Reply:
and this is my migrate for adding confirmable to devise.

![add confirm](https://user-images.githubusercontent.com/96838030/188913165-f7c7a97f-8748-4e58-8999-1e1f6e26f085.png)

Reply:
Hi @ahzamir

It could be that you need to permit a `confirmable password` in your Device parameters.
Kindly try to add a `protected` method `update_allowed_parameters` in your `application_controller.rb` and make sure that you permit `:password_confirmable` on signing up as illustrated below:

![image](https://user-images.githubusercontent.com/38088818/188912609-47efd0bb-6270-4cf0-b756-f7d277fd971b.png)

Here's a code snippet for easy copy:
```
  def update_allowed_parameters
    devise_parameter_sanitizer.permit(:sign_up) do |u|
      u.permit(:name, :posts_counter, :email, :password, :password_confirmable)
    end
  end
```

The second permit code is OPTIONAL, you can include it if you intend to also have account updates to allow users to change their password or other related information in your app.
```
    devise_parameter_sanitizer.permit(:account_update) do |u|
      u.permit(:name, :surname, :email, :password, :current_password)
    end
```

Reply:
@KDlamini  Thank you very much 
- I have tried this way and I added the mentioned method and its `permit()`.
Then I run `$ rake db:drop db:create db:migrate` and restart my `rails s` and when I sign up I got this error:

![error 3](https://user-images.githubusercontent.com/96838030/188919023-ca116879-9b05-4015-b5a2-b1c83abb1338.png)

- And this is my `application_controller.rb` file:

![application record](https://user-images.githubusercontent.com/96838030/188919144-61696ab7-90ed-4742-9265-4da888680533.png)

- This is the schema for my user class:

![schema user](https://user-images.githubusercontent.com/96838030/188919330-2d339534-b5ff-4344-a068-94c12c91e38d.png)

Reply:
@KDlamini 
- And in addition to the last comment when I click the `[Didn't receive confirmation instructions?]` which is directed to `(http://localhost:3000/users/confirmation/new)` I get this error:

![confirmation not found](https://user-images.githubusercontent.com/96838030/188924734-58303d81-275c-42f7-adfe-1d227fe6e0d8.png)

Reply:
@ahzamir Please check the instructions I sent you on Slack. You should use the migration file created by Device to add the confirmation attributes to your table.